detectCycleInLinkedList.js

```diff
@@ -0,0 +1,14 @@
+var hasCycle = function (head) {
```

Comment:
Never recommended to increment head, because that is the only reference you have to scan the original linked list.

Take a variable slow = head;